src/helpers/figures/charts/scorecard_chart_designer.ts

```diff
@@ -0,0 +1,377 @@
+import { Color } from "chart.js";
+import { DEFAULT_FONT } from "../../../constants";
+import { Figure, Pixel, PixelPosition, Style } from "../../../types";
+import { ScorecardChartRuntime } from "../../../types/chart";
+import { relativeLuminance } from "../../color";
+import { getFontSizeMatchingWidth } from "../../text_helper";
+
+/* Sizes of boxes containing the texts, in percentage of the Chart size */
+const TITLE_FONT_SIZE = 18;
+
+const BASELINE_BOX_HEIGHT_RATIO = 0.35;
+const KEY_BOX_HEIGHT_RATIO = 0.65;
+
+/** Baseline description should have a smaller font than the baseline */
+const BASELINE_DESCR_FONT_RATIO = 0.9;
+
+/* Padding at the border of the chart, in percentage of the chart width */
+const CHART_PADDING_RATIO = 0.02;
+
+/**
+ * Line height (in em)
+ * Having a line heigh =1em (=font size) don't work, the font will overflow.
+ */
+const LINE_HEIGHT = 1.2;
+
+type ScorecardChartDesignElement = {
+  text: string;
+  style: {
+    font: string;
+    color: Color;
+    strikethrough?: boolean;
+    underline?: boolean;
+  };
+  position: PixelPosition;
+};
+
+export type ScorecardChartDesign = {
```

Comment:
Now there are 3 data structures: definition, runtime and this. There's no clear difference between the last two. And there's no clear reason why they both exists.
Runtime should become this new structure.
And all this should be moved to `createScorecardChartRuntime`

Reply:
@anhe-odoo @hokolomopo 
We were discussing in private with @rrahir about that. He has a point about this split.
- it's true that runtime and designer being different things can make sense (one depends on the model, gets the colors, texts, etc., the other takes that structure and computes a more specialized structure with positions to draw on the canvas)
- however, this is the same thing for other (chart.js) charts: the runtime is already this specialized structure ready.

For consistency with other charts (chart.js), it's better to keep every flow (basically) the same. It will help the reader to understand how it works: he understands the flow for one chart type, he understands all charts :)

Merging scorecard runtime and designer doesn't mean we can't keep things neat and tidy in `createScorecardChartRuntime`

What do you think?

Reply:
nitpick ,the name kinda collides with ScorecardChartDesignPanel